### PR TITLE
refactor log handling

### PR DESCRIPTION
**Pull request by jkhelil**
# Changes

I propose avoid passing logger as argument in function signature because it is confusing, adding complexity.
we can either add and init the logger in the struct object in order to use it on underlying method or just add it in method 

for example:
return &ReconcileJenkinsBaseConfiguration{
		Configuration:                config,
		logger:                       log.Log.WithValues("cr", config.Jenkins.Name),
		jenkinsAPIConnectionSettings: jenkinsAPIConnectionSettings,
	}
or here:
func (r *ReconcileJenkins) Reconcile(request reconcile.Request) (reconcile.Result, error) {
	reconcileFailLimit := uint64(10)
	logger := logx.WithValues("cr", request.Name)
	

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes tests (if functionality changed/added)
- [ ] Includes docs (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/jenkinsci/kubernetes-operator/blob/master/CONTRIBUTING.md#commit-messages)

_See [the contribution guide](https://github.com/jenkinsci/kubernetes-operator/blob/master/CONTRIBUTING.md) for more details._

## Reviewer Notes

If API changes are included, additive changes must be approved by at least two [OWNERS](https://github.com/jenkinsci/kubernetes-operator/blob/master/OWNERS) and backwards incompatible changes must be approved by [more than 50% of the OWNERS](https://github.com/jenkinsci/kubernetes-operator/blob/master/OWNERS).

# Release Notes

```
Describe any user facing changes here, or delete this block.

Examples of user facing changes:
- API changes
- Bug fixes
- Any changes in behavior

```

### CODE REVIEW

File: pkg/controller/jenkins/jenkins_controller.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func (r *ReconcileJenkins) Reconcile(request reconcile.Request) (reconcile.Resul
 		reconcileErrors[request.Name] = lastErrors
 		if lastErrors.counter >= reconcileFailLimit {
 			if log.Debug {
-				logger.V(log.VWarn).Info(fmt.Sprintf("Reconcile loop failed %d times with the same error, giving up: %+v", reconcileFailLimit, err))
+				logger.V(log.VDebug).Info(fmt.Sprintf("Reconcile loop failed %d times with the same error, giving up: %+v", reconcileFailLimit, err))
```
Comment (tomaszsek) [inline pkg/controller/jenkins/jenkins_controller.go:151]: please revert this line as it is warn not debug

File: pkg/controller/jenkins/jenkins_controller.go [hunk 2]
```diff
@@ -166,7 +166,7 @@ func (r *ReconcileJenkins) Reconcile(request reconcile.Request) (reconcile.Resul
 		}
 
 		if log.Debug {
-			logger.V(log.VWarn).Info(fmt.Sprintf("Reconcile loop failed: %+v", err))
+			logger.V(log.VDebug).Info(fmt.Sprintf("Reconcile loop failed: %+v", err))
```
Comment (tomaszsek) [inline pkg/controller/jenkins/jenkins_controller.go:169]: please revert this line as it is warn not debug

Comment (jkhelil) [reply pkg/controller/jenkins/jenkins_controller.go:169]: So what it is the purpose of the test ? the two lines are identical

File: pkg/controller/jenkins/jenkins_controller.go (hunk 1 above)
Comment (jkhelil) [reply pkg/controller/jenkins/jenkins_controller.go:151]: So what it is the purpose of the test ? the two lines are identical

File: pkg/controller/jenkins/jenkins_controller.go (hunk 2 above)
Comment (jkhelil) [reply pkg/controller/jenkins/jenkins_controller.go:169]: can we simplify by removing the test here
if log.Debug {
			logger.V(log.VWarn).Info(fmt.Sprintf("Reconcile loop failed: %+v", err))
		} else if err.Error() != fmt.Sprintf("Operation cannot be fulfilled on jenkins.jenkins.io \"%s\": the object has been modified; please apply your changes to the latest version and try again", request.Name) {
			logger.V(log.VWarn).Info(fmt.Sprintf("Reconcile loop failed: %s", err))
		}

Comment (jkhelil) [reply pkg/controller/jenkins/jenkins_controller.go:169]: Seems that in most all cases we log the same thing
logger.V(log.VWarn).Info(fmt.Sprintf("Reconcile loop failed: %s", err))

Comment (tomaszsek) [reply pkg/controller/jenkins/jenkins_controller.go:169]: We can't remove the if here, please look at `%+v` vs `%s`. If debug is enabled full stack trace will be printed.

Comment (jkhelil) [reply pkg/controller/jenkins/jenkins_controller.go:169]: ok I see thanks

File: pkg/controller/jenkins/jenkins_controller.go (hunk 1 above)
Comment (tomaszsek) [reply pkg/controller/jenkins/jenkins_controller.go:151]:
```suggestion
				logger.V(log.VWarn).Info(fmt.Sprintf("Reconcile loop failed %d times with the same error, giving up: %+v", reconcileFailLimit, err))
```

File: pkg/controller/jenkins/jenkins_controller.go (hunk 2 above)
Comment (tomaszsek) [reply pkg/controller/jenkins/jenkins_controller.go:169]:
```suggestion
			logger.V(log.VWarn).Info(fmt.Sprintf("Reconcile loop failed: %+v", err))
```

Comment (jkhelil) [reply pkg/controller/jenkins/jenkins_controller.go]: fixed

**Pull request merged**